cmd/kubeadm/app/cmd/join_test.go

```diff
@@ -312,6 +313,8 @@ func TestNewJoinData(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			// initialize an external join option and inject it to the join cmd
 			joinOptions := newJoinOptions()
+			// set the controlPlane to true to bypass the discovery of a tlsBootstrapCfg from the JoinConfiguration, this could be opt-out by the flag defined in the testcases.
+			joinOptions.controlPlane = true
```

Comment:
can you please elaborate on why this controlPlane field bypasses the discovery of a tlsBootstrapCfg? is it because the tlsBootstrapCfg local var will be nil if there is no admin.conf?
i got a bit lost on this one. maybe more details in the comment are needed.

Reply:
overall, we want to bypass the logic to get the `tlsBootstrapCfg` from a cluster, unittest shouldn't do that (interact with a cluster to fetch something from a cluster).

https://github.com/kubernetes/kubernetes/pull/119946/files#diff-c6f6a594b0b45b48a444c274731731e10883c6cf9e2e66f8938aa28af5a57b27R436-R437

```golang
	if tlsBootstrapCfg == nil {
		if tlsBootstrapCfg, err = discovery.For(cfg); err == nil {
```


set the controlplane will help to set `tlsBootstrapCfg` and hence skip this step.